File: oss/mcp-tooling/workers/src/routes/enterprise.ts
import { Hono } from 'hono';
import { zValidator } from '@hono/zod-validator';
import { z } from 'zod';
import { paymentService } from '../services/payment';
import { auditService } from '../services/audit';
import { authMiddleware, type Env } from '../middleware/auth';

const enterpriseRouter = new Hono<{ Bindings: Env }>();

// Request Demo
enterpriseRouter.post(
    '/demo',
    zValidator(
        'json',
        z.object({
            first_name: z.string(),
            last_name: z.string(),
            email: z.string().email(),
            company: z.string(),
            description: z.string().optional(),
        })
    ),
    async (c) => {
        const body = c.req.valid('json');

        // In reality: Trigger CRM workflow, email sales team, etc.
        return c.json({
            message: 'Demo request received',
            id: `demo_req_${Date.now()}`
        });
    }
);

// Get Dedicated Config
enterpriseRouter.get('/dedicated-config', (c) => {
    return c.json({
        regions: [
            'us-east-1-dedicated',
            'eu-west-1-dedicated',
        ],
        instance_types: [
            'm5.large',
            'm5.xlarge',
            'c5.xlarge',
        ],
        isolation_levels: [
            'pod',
            'node',
            'cluster',
        ],
    });
});

// Create Payment Intent (requires auth)
enterpriseRouter.post(
    '/payments/intent',
    authMiddleware,
    zValidator(
        'json',
        z.object({
            user_email: z.string().email(),
            variant_id: z.string(),
            amount: z.number().optional()
        })
    ),
    async (c) => {
        const body = c.req.valid('json');

        const params = {
            user_email: body.user_email,
            amount: body.amount || 0,
            currency: 'USD',
            variant_id: body.variant_id
        };

        try {
            const result = await paymentService.createPayment(params);

            // Audit the transaction
            await auditService.log(c, 'create_payment_intent', 'payment', {
                amount: params.amount,
                variant_id: params.variant_id
            });

            return c.json(result);
        } catch (err: any) {
            return c.json({ error: err.message }, 500);
        }
    }
);

export { enterpriseRouter };

File: oss/mcp-tooling/services/api-service/internal/handlers/enterprise.go
package handlers

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"github.com/mcpoverflow/api-service/internal/config"
	"github.com/mcpoverflow/api-service/internal/services"
)

type EnterpriseHandler struct {
	paymentService *services.PaymentService
	cfg            *config.Config
}

func NewEnterpriseHandler(cfg *config.Config) *EnterpriseHandler {
	return &EnterpriseHandler{
		paymentService: services.NewPaymentService(),
		cfg:            cfg,
	}
}

// RequestDemo handles enterprise demo requests
func (h *EnterpriseHandler) RequestDemo(c *gin.Context) {
	var req struct {
		FirstName   string `json:"first_name" binding:"required"`
		LastName    string `json:"last_name" binding:"required"`
		Email       string `json:"email" binding:"required,email"`
		Company     string `json:"company" binding:"required"`
		Description string `json:"description"`
	}

	if err := c.ShouldBindJSON(&req); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}

	// In reality: Trigger CRM workflow, email sales team, etc.
	c.JSON(http.StatusOK, gin.H{
		"message": "Demo request received",
		"id":      "demo_req_12345",
	})
}

// CreatePaymentIntent handles payment intent creation
func (h *EnterpriseHandler) CreatePaymentIntent(c *gin.Context) {
	var req services.CreatePaymentParams
	if err := c.ShouldBindJSON(&req); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}

	result, err := h.paymentService.CreatePayment(c.Request.Context(), req)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}

	c.JSON(http.StatusOK, result)
}

// GetDedicatedConfig returns configuration for dedicated deployments
func (h *EnterpriseHandler) GetDedicatedConfig(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{
		"regions": []string{
			"us-east-1-dedicated",
			"eu-west-1-dedicated",
		},
		"instance_types": []string{
			"m5.large",
			"m5.xlarge",
			"c5.xlarge",
		},
		"isolation_levels": []string{
			"pod",
			"node",
			"cluster",
		},
	})
}
